dotCMS/src/main/java/com/dotmarketing/portlets/fileassets/business/FileAssetAPIImpl.java

```diff
@@ -304,35 +292,45 @@ public boolean fileNameExists(Host host, Folder folder, String fileName, String
 			return false;
 		}
 
-		boolean exist = false;
-
-		Identifier folderId = APILocator.getIdentifierAPI().find(folder);
-		String path = folder.getInode().equals(FolderAPI.SYSTEM_FOLDER)?"/"+fileName:folderId.getPath()+fileName;
-		Identifier fileAsset = APILocator.getIdentifierAPI().find(host, path);
-
-		if(fileAsset!=null && InodeUtils.isSet(fileAsset.getId()) && !identifier.equals(fileAsset.getId()) && !fileAsset.getAssetType().equals("folder")){
-			// Let's not break old logic. ie calling fileNameExists method without languageId parameter.
-			if (languageId == -1){
-				exist = true;
-			} else { // New logic.
-				//We need to make sure that the contentlets for this identifier have the same language.
-				try {
-					contAPI.findContentletByIdentifier(fileAsset.getId(), false, languageId,
-						APILocator.getUserAPI().getSystemUser(), false);
-					exist = true;
-				} catch (DotSecurityException dse) {
-					// Something could failed, lets log and assume true to not break anything.
-					Logger.error(FileAssetAPIImpl.class,
-						"Error trying to find contentlet from identifier:" + fileAsset.getId(), dse);
-					exist = true;
-				} catch (DotContentletStateException dcse){
-					// DotContentletStateException is thrown when content is not found.
-					exist = false;
+		final Identifier folderId = APILocator.getIdentifierAPI().find(folder);
+		final String path = folder.getInode().equals(FolderAPI.SYSTEM_FOLDER) ? StringPool.FORWARD_SLASH + fileName : folderId.getPath() + fileName;
+		final Identifier fileAssetIdentifier = APILocator.getIdentifierAPI().find(host, path);
+		if(( null == fileAssetIdentifier || !InodeUtils.isSet(fileAssetIdentifier.getId()) || "folder".equals(fileAssetIdentifier.getAssetType()))){
+		   // if we're looking at a folder or the fileAssetIdentifier wasn't found. It doesn't exist for sure.
+		   return false;
+		}
+            //Beyond this point we know something matches that path for that host.
+			if(!UtilMethods.isSet(identifier)){
+			   //it's a brand new contentlet we're dealing with
+			   //At this point we know it DOES exist, and since we're dealing with a fresh contentlet that hasn't even been inserted yet (We don't need to worry about lang).
+			   return true;
+			} else {
+				// old logic. ie calling fileNameExists method without languageId parameter.
+				if (languageId == -1){
```

Comment:
Why do we care about the language, `languageId` should not even be a parameter, we just care about what is in the identifier table right?